src/domain/RBF/rbfSupport.cpp

```diff
@@ -0,0 +1,465 @@
+        if (closure[cl] >= vStart && closure[cl] < vEnd) {                                       // Only use the points corresponding to either a vertex or edge/face.
+            PetscCall(DMPlexGetTransitiveClosure(dm, closure[cl], PETSC_FALSE, &nStar, &star));  // Get all points using this vertex or edge/face.
+
+            for (st = 0; st < nStar * 2; st += 2) {
+                if (star[st] >= cStart && star[st] < cEnd) {                               // If the point is a cell add it.
+                    PetscCall(DMPlexComputeCellGeometryFVM(dm, star[st], NULL, x, NULL));  // Center of the candidate cell.
+                    dist = 0.0;
+                    for (i = 0; i < dim; ++i) {  // Compute the distance so that we can check if it's within the required distance.
+                        dist += PetscSqr(x0[i] - x[i]);
+                    }
+
+                    if (dist <= maxDist && star[st] != p) {  // Only add if the distance is within maxDist and it's not the center cell
+                        list[n++] = star[st];
+                    }
+                }
+            }
+
+            PetscCall(DMPlexRestoreTransitiveClosure(dm, closure[cl], PETSC_FALSE, &nStar, &star));  // Restore the points
+        }
+    }
+
+    PetscCall(DMPlexRestoreTransitiveClosure(dm, p, PETSC_TRUE, &nClosure, &closure));  // Restore the points
+    PetscCall(PetscSortRemoveDupsInt(&n, list));                                        // Cleanup the list
+    if (!(*cells)) PetscCall(PetscMalloc1(n, cells));                                   // Allocate the output
+    PetscCall(PetscArraycpy(*cells, list, n));                                          // Copy the cell list
+    *nCells = n;                                                                        // Set the number of cells for output
+
+    PetscFunctionReturn(0);
+}
+
+// Return all values in sorted array a that are NOT in sorted array b. This is done in-place on array a.
+PetscErrorCode PetscSortedArrayComplement(const PetscInt nb, const PetscInt b[], PetscInt *na, PetscInt a[]) {
+    PetscFunctionBegin;
+
+    PetscInt i = 0, j = 0;
+    PetscInt n = 0;
+
+    while (i < *na && j < nb) {
+        if (a[i] < b[j]) {  // If the current element in b[] is larger, therefore a[i] cannot be included in b[j..p-1].
+            a[n++] = a[i];
+            i++;
+        } else if (a[i] > b[j]) {  // Smaller elements of b[] are skipped
+            j++;
+        } else if (a[i] == b[j]) {  // Same elements detected (skipping in both arrays)
+            i++;
+            j++;
+        }
+    }
+
+    // If a[] is larger than b[] then all remaining values must be unique as this is a sorted array.
+    for (; i < *na; ++i) {
+        a[n++] = a[i];
+    }
+
+    // Assign the number of unique values
+    *na = n;
+
+    PetscFunctionReturn(0);
+}
+
+// Return the list of neighboring cells to cell p using a combination of number of levels and maximum distance
```

Comment:
I'm not sure if these comments will show up in the doxygen https://www.doxygen.nl/manual/docblocks.html .